### PR TITLE
Move PROXY_TO_PTHREAD handling into native code. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #17144

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -9193,6 +9193,15 @@ def test_pthread_dylink_longjmp(self):
...
     self.dylink_testf(main, need_reverse=False)
 
+  @needs_dylink
+  @node_pthreads
+  def test_pthread_dylink_main_module_1(self):
+    self.emcc_args.append('-Wno-experimental')
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('USE_PTHREADS')
+    self.set_setting('MAIN_MODULE')
+    self.do_runf(test_file('hello_world.c'))
```
Comment (tlively) [inline tests/test_core.py:9203]: Is this test related?

**Review by tlively — APPROVED** [review]

File: tests/test_core.py (hunk 1 above)
Comment (sbc100) [reply tests/test_core.py:9205]: Yes, this test fails before this change since with `__call_main` being undefined.

File: system/lib/pthread/emscripten_proxy_main.c [hunk 2]
```diff
@@ -14,12 +15,13 @@
...
 
-extern void __call_main(int argc, char** argv);
+extern int main(int argc, char** argv);
 
 static void* _main_thread(void* param) {
   // This is the main runtime thread for the application.
   emscripten_set_thread_name(pthread_self(), "Application main thread");
-  __call_main(_main_argc, _main_argv);
+  int rtn = main(_main_argc, _main_argv);
+  exit(rtn);
```
Comment (dschuff) [inline system/lib/pthread/emscripten_proxy_main.c:24]: I guess this is what replaces the logic above of exiting on the main thread?
